step-agent/src/main/java/step/grid/agent/AgentRunner.java

```diff
@@ -40,6 +40,7 @@ public static void main(String[] args) throws Exception {
 		Runtime.getRuntime().addShutdownHook(new Thread(()->{
 			try {
 				agent.close();
+				cl.close();

```

Comment:
If the cl has to be closed in the case of an error while closing the agent, we should move this to a dedicated try block